From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Paolo <paolo.sabatino@gmail.com>
Date: Fri, 19 Jun 2020 17:27:27 +0200
Subject: [ARCHEOLOGY] Introducing Rockchip RK322X SoC support (#2032)

> X-Git-Archeology: > recovered message: > * Introducing Rockchip rk322x SoC support
> X-Git-Archeology: > recovered message: > Main features:
> X-Git-Archeology: > recovered message: > - Legacy kernel flavour based upon stable v2.x rk3288 Rockchip branch (https://github.com/rockchip-linux/kernel/tree/stable-4.4-rk3288-linux-v2.x)
> X-Git-Archeology: > recovered message: > - Current kernel flavour based on mainline 5.6.y kernel
> X-Git-Archeology: > recovered message: > - Mainline u-boot (v2020.04)
> X-Git-Archeology: > recovered message: > - Single generic tv box target (rk322x-box) which boots on all the known tv boxes
> X-Git-Archeology: > recovered message: > - Hardware devices (eMMC/NAND, led wiring configuration, SoC variant selection) modulation done by user at runtime via device tree overlays - a script (rk322x-config) is provided for autodetection and simple configuration by inexperienced users;
> X-Git-Archeology: > recovered message: > - Bits added to armbian-hardware-optimization to set affinity for irq handlers
> X-Git-Archeology: > recovered message: > - rk322x-box targets already added to targets.conf for automatic image creation
> X-Git-Archeology: > recovered message: > * Removed disabled patches
> X-Git-Archeology: > recovered message: > * Restored mysteriously removed comment character
> X-Git-Archeology: - Revision 23604e8a0dcdf81ec6c28ccd4b2a64b90816d8e7: https://github.com/armbian/build/commit/23604e8a0dcdf81ec6c28ccd4b2a64b90816d8e7
> X-Git-Archeology:   Date: Fri, 19 Jun 2020 17:27:27 +0200
> X-Git-Archeology:   From: Paolo <paolo.sabatino@gmail.com>
> X-Git-Archeology:   Subject: Introducing Rockchip RK322X SoC support (#2032)
> X-Git-Archeology:
> X-Git-Archeology: - Revision bd17d4dbd0025908b6f0aa58d74f2bc8cfedb076: https://github.com/armbian/build/commit/bd17d4dbd0025908b6f0aa58d74f2bc8cfedb076
> X-Git-Archeology:   Date: Sat, 19 Sep 2020 15:20:16 +0000
> X-Git-Archeology:   From: paolo <paolo.sabatino@gmail.com>
> X-Git-Archeology:   Subject: Many changes for rk322x target:
> X-Git-Archeology:
> X-Git-Archeology: - Revision 95425c27b9d3bbb96e7936cc531638c9150538f9: https://github.com/armbian/build/commit/95425c27b9d3bbb96e7936cc531638c9150538f9
> X-Git-Archeology:   Date: Fri, 12 Mar 2021 20:20:12 +0000
> X-Git-Archeology:   From: Paolo Sabatino <paolo.sabatino@gmail.com>
> X-Git-Archeology:   Subject: Changes and fixes to rk322x uboot and kernel config
> X-Git-Archeology:
> X-Git-Archeology: - Revision 0cdffb29b07305209efb12cf3b5ac6032d3a1153: https://github.com/armbian/build/commit/0cdffb29b07305209efb12cf3b5ac6032d3a1153
> X-Git-Archeology:   Date: Wed, 24 Mar 2021 19:01:53 +0100
> X-Git-Archeology:   From: Igor Pecovnik <igorpecovnik@users.noreply.github.com>
> X-Git-Archeology:   Subject: Renaming DEV branch to EDGE (#2704)
> X-Git-Archeology:
> X-Git-Archeology: - Revision 4c3dcbf4fcd3616999cb91a1dddfa74668eb6de9: https://github.com/armbian/build/commit/4c3dcbf4fcd3616999cb91a1dddfa74668eb6de9
> X-Git-Archeology:   Date: Tue, 09 Nov 2021 21:58:35 +0100
> X-Git-Archeology:   From: Paolo <paolo.sabatino@gmail.com>
> X-Git-Archeology:   Subject: Rockchip 5.15 (#3242)
> X-Git-Archeology:
> X-Git-Archeology: - Revision 0a2892aeba4dec4633a86f8ad573bf62f09cdb38: https://github.com/armbian/build/commit/0a2892aeba4dec4633a86f8ad573bf62f09cdb38
> X-Git-Archeology:   Date: Sun, 24 Sep 2023 19:47:01 +0200
> X-Git-Archeology:   From: Paolo Sabatino <paolo.sabatino@gmail.com>
> X-Git-Archeology:   Subject: rk322x: bump edge kernel to 6.5
> X-Git-Archeology:
> X-Git-Archeology: - Revision 562d96128ba6a511a8a06c0f4d29946ab80b8969: https://github.com/armbian/build/commit/562d96128ba6a511a8a06c0f4d29946ab80b8969
> X-Git-Archeology:   Date: Tue, 26 Dec 2023 16:45:30 +0100
> X-Git-Archeology:   From: Paolo Sabatino <paolo.sabatino@gmail.com>
> X-Git-Archeology:   Subject: consolidate rk322x and rockchip 32 bit families
> X-Git-Archeology:
> X-Git-Archeology: - Revision 54628d7d3e11824e560b77e905f69d52feb0fbd0: https://github.com/armbian/build/commit/54628d7d3e11824e560b77e905f69d52feb0fbd0
> X-Git-Archeology:   Date: Wed, 01 Jan 2025 19:38:55 +0100
> X-Git-Archeology:   From: Paolo Sabatino <paolo.sabatino@gmail.com>
> X-Git-Archeology:   Subject: rockchip: bump edge kernel to 6.13-rc5
> X-Git-Archeology:
> X-Git-Archeology: - Revision 7c55b4fce91f38383398a7498dde1c6d69a70495: https://github.com/armbian/build/commit/7c55b4fce91f38383398a7498dde1c6d69a70495
> X-Git-Archeology:   Date: Wed, 26 Mar 2025 22:23:29 +0100
> X-Git-Archeology:   From: Paolo Sabatino <paolo.sabatino@gmail.com>
> X-Git-Archeology:   Subject: bump rockchip 32bit edge kernel to v6.14
> X-Git-Archeology:
> X-Git-Archeology: - Revision cc4cb72d4069147ea1b5e6936de3b49aace21967: https://github.com/armbian/build/commit/cc4cb72d4069147ea1b5e6936de3b49aace21967
> X-Git-Archeology:   Date: Tue, 03 Jun 2025 09:53:37 +0200
> X-Git-Archeology:   From: Paolo Sabatino <paolo.sabatino@gmail.com>
> X-Git-Archeology:   Subject: bump rockchip 32 bit edge kernel to 6.15
> X-Git-Archeology:
> X-Git-Archeology: - Revision 96fe7dee19eaec6d9c5159a5cc50e33ca9c96096: https://github.com/armbian/build/commit/96fe7dee19eaec6d9c5159a5cc50e33ca9c96096
> X-Git-Archeology:   Date: Mon, 28 Jul 2025 20:45:52 +0800
> X-Git-Archeology:   From: Paolo Sabatino <paolo.sabatino@gmail.com>
> X-Git-Archeology:   Subject: bump rockchip edge to kernel 6.16
> X-Git-Archeology:
> X-Git-Archeology: - Revision 29317c6f7e33f2cc509acc0da23b615a7d9d8c31: https://github.com/armbian/build/commit/29317c6f7e33f2cc509acc0da23b615a7d9d8c31
> X-Git-Archeology:   Date: Thu, 18 Sep 2025 22:48:06 +0200
> X-Git-Archeology:   From: Paolo Sabatino <paolo.sabatino@gmail.com>
> X-Git-Archeology:   Subject: update rockchip 32 bit edge kernel to 6.17
> X-Git-Archeology:
---
 arch/arm/boot/dts/rockchip/rk322x.dtsi | 75 ++++++++++
 1 file changed, 75 insertions(+)

diff --git a/arch/arm/boot/dts/rockchip/rk322x.dtsi b/arch/arm/boot/dts/rockchip/rk322x.dtsi
index 111111111111..222222222222 100644
--- a/arch/arm/boot/dts/rockchip/rk322x.dtsi
+++ b/arch/arm/boot/dts/rockchip/rk322x.dtsi
@@ -817,6 +817,22 @@ emmc: mmc@30020000 {
 		status = "disabled";
 	};
 
+	nfc: nand-controller@30030000 {
+                compatible = "rockchip,rk3228-nfc", "rockchip,rk2928-nfc";
+                reg = <0x30030000 0x4000>;
+                interrupts = <GIC_SPI 15 IRQ_TYPE_LEVEL_HIGH>;
+                clocks = <&cru SCLK_NANDC>, <&cru HCLK_NANDC>;
+                clock-names = "nfc", "ahb";
+                assigned-clocks = <&cru SCLK_NANDC>;
+                assigned-clock-rates = <150000000>;
+
+                pinctrl-names = "default";
+                pinctrl-0 = <&flash_cs0 &flash_rdy &flash_ale &flash_cle
+                        &flash_wrn &flash_rdn &flash_bus8>;
+                status = "disabled";
+
+        };
+
 	usb_otg: usb@30040000 {
 		compatible = "rockchip,rk3228-usb", "rockchip,rk3066-usb",
 			     "snps,dwc2";
@@ -1100,6 +1116,65 @@ emmc_bus8: emmc-bus8 {
 			};
 		};
 
+		flash {
+
+                        flash_cs0: flash-cs0 {
+                                rockchip,pins = <2 RK_PA6 1 &pcfg_pull_up>;
+                        };
+
+                        flash_cs1: flash-cs1 {
+                                rockchip,pins = <0 RK_PC7 1 &pcfg_pull_up>;
+                        };
+
+                        flash_cs2: flash-cs2 {
+                                rockchip,pins = <1 RK_PC6 1 &pcfg_pull_up>;
+                        };
+
+                        flash_cs3: flash-cs3 {
+                                rockchip,pins = <1 RK_PC7 1 &pcfg_pull_up>;
+                        };
+
+                        flash_rdy: flash-rdy {
+                                rockchip,pins = <2 RK_PA4 1 &pcfg_pull_up>;
+                        };
+
+                        flash_ale: flash-ale {
+                                rockchip,pins = <2 RK_PA0 1 &pcfg_pull_down>;
+                        };
+
+                        flash_cle: flash-cle {
+                                rockchip,pins = <2 RK_PA1 1 &pcfg_pull_down>;
+                        };
+
+                        flash_wrn: flash-wrn {
+	                        rockchip,pins = <2 RK_PA2 1 &pcfg_pull_up>;
+                        };
+
+                        flash_rdn: flash-rdn {
+                                rockchip,pins = <2 RK_PA3 1 &pcfg_pull_up>;
+                        };
+
+			flash_bus8: flash-bus8 {
+                                rockchip,pins = <1 RK_PD0 1 &pcfg_pull_up>,
+                                                <1 RK_PD1 1 &pcfg_pull_up>,
+                                                <1 RK_PD2 1 &pcfg_pull_up>,
+                                                <1 RK_PD3 1 &pcfg_pull_up>,
+                                                <1 RK_PD4 1 &pcfg_pull_up>,
+                                                <1 RK_PD5 1 &pcfg_pull_up>,
+                                                <1 RK_PD6 1 &pcfg_pull_up>,
+                                                <1 RK_PD7 1 &pcfg_pull_up>;
+                        };
+
+                        flash_dqs: flash-dqs {
+                                rockchip,pins = <2 RK_PA7 1 &pcfg_pull_up>;
+                        };
+
+                        flash_wp: flash-wp {
+                                rockchip,pins = <2 RK_PA5 1 &pcfg_pull_down>;
+                        };
+
+                };
+
 		gmac {
 			rgmii_pins: rgmii-pins {
 				rockchip,pins = <2 RK_PB6 1 &pcfg_pull_none>,
-- 
Armbian

